jena-core/src/main/java/org/apache/jena/irix/IRIs.java

```diff
@@ -76,24 +76,32 @@ public static String getBaseStr() {
     /**
      * Given a candidate baseURI string, which may be a filename,
      * turn it into a IRI suitable as a base IRI.
+     * This includes encoding characters in a filename (e.g. spaces).
      */
-    public static String toBase(String baseURI) {
-        if ( baseURI == null )
+    public static String toBase(String uriForBase) {
+        if ( uriForBase == null )
             return getBaseStr();
-        String scheme = scheme(baseURI);
+        String scheme = scheme(uriForBase);
         if ( Sys.isWindows ) {
             // Assume a scheme of one letter is a Windows drive letter.
             if ( scheme != null && scheme.length() == 1 )
                 scheme = "file";
         }
-        // File scheme if the URI is "file:" or there is no scheme
-        // and the system base is a "file:" URI
-        boolean isFile = ( scheme != null )
-                ? scheme.equals("file")
-                : IRIs.getSystemBase().hasScheme("file");
-        if ( isFile )
-            baseURI = IRILib.encodeFileURL(baseURI);
-        return IRIs.getSystemBase().resolve(baseURI).toString();
+        if ( scheme == null  ) {
+            // Relative name: it the base is a file: URI, encode the relative
+            // name if it does not look like it is already encoded.
+            boolean isFileBase = ( scheme == null ) && IRIs.getSystemBase().hasScheme("file");
```

Comment:
This `( scheme == null )`  will always be `true`, because of the `if ( scheme == null )` above.

Reply:
Fixed!